src/fw-logs/fw-log-data.cpp

```diff
@@ -1,68 +1,63 @@
 // License: Apache 2.0. See LICENSE file in root directory.
-// Copyright(c) 2019 Intel Corporation. All Rights Reserved.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
 #include "fw-log-data.h"
-#include <sstream>
-#include <iomanip>
-#include <locale>
-#include <string>
 
-using namespace std;
+#include <rsutils/string/from.h>
 
-# define SET_WIDTH_AND_FILL(num, element) \
-setfill(' ') << setw(num) << left << element \
 
 namespace librealsense
 {
     namespace fw_logs
     {
-        rs2_log_severity fw_log_data::get_severity() const
-        {
-            return fw_logs_severity_to_log_severity(_severity);
-        }
-
-        const std::string& fw_log_data::get_message() const
-        {
-            return _message;
-        }
-
-        const std::string& fw_log_data::get_file_name() const
-        {
-            return _file_name;
-        }
-
-        const std::string& fw_log_data::get_thread_name() const
-        {
-            return _thread_name;
-        }
-
-        uint32_t fw_log_data::get_line() const
-        {
-            return _line;
-        }
-
-        uint32_t fw_log_data::get_timestamp() const
+        rs2_log_severity fw_logs_binary_data::get_severity() const
         {
-            return (uint32_t)_timestamp;
-        }
+            const fw_log_binary * log_binary = reinterpret_cast< const fw_log_binary * >( logs_buffer.data() );
+            if( log_binary->magic_number == 0xA5)
```

Comment:
This is how we distinguish d400 from d500?
magic number was changed?

Reply:
We need to support `get_severity` from the log entry binary data, in case no parser is available.
This is the easiest way I found doing so.